### PR TITLE
LPS-148805

### PR DESCRIPTION
**Pull request by leticia-maciel**

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/persistence/impl/DDMFormInstanceFinderImpl.java [hunk 1]
```diff
@@ -263,6 +307,16 @@ protected int doCountByC_G_N_D_S(
 					sql, "(DDMFormInstance.groupId = ?) AND");
 			}
 
+			if (published == null) {
+				sql = StringUtil.removeSubstring(
+					sql, "(DDMFormInstance.settings_ LIKE '%\"fieldReference\":\"published\",\"value\":\"[$PUBLISHED$]\"%') AND");
```
Comment (leticia-maciel) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/persistence/impl/DDMFormInstanceFinderImpl.java:312]: SF does not work. Long line.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/persistence/impl/DDMFormInstanceFinderImpl.java [hunk 2]
```diff
@@ -263,6 +307,16 @@ protected int doCountByC_G_N_D_S(
 					sql, "(DDMFormInstance.groupId = ?) AND");
 			}
 
+			if (published == null) {
```
Comment (leticia-maciel) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/persistence/impl/DDMFormInstanceFinderImpl.java:310]: May return error.

**Comment by leticia-maciel** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/persistence/impl/DDMFormInstanceFinderImpl.java [hunk 3]
```diff
@@ -148,15 +157,27 @@ public List<DDMFormInstance> filterFindByC_G_N_D(
 			orderByComparator, true);
 	}
 
+	@Override
+	public List<DDMFormInstance> filterFindByC_G_N_D_P(
```
Comment (rodrigopaulino) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/persistence/impl/DDMFormInstanceFinderImpl.java:161]: I don't think we need this method, @leticia-maciel. We can call `doFindByC_G_N_D_P_S` directly in line 476.

**Review by rodrigopaulino — CHANGES_REQUESTED** [review]
Hey @leticia-maciel, can you please take a look at my comment below? Thanks!

**Comment by leticia-maciel** [comment]
ci:test:relevant

**Comment by leticia-maciel** [comment]
ci:test:relevant

**Review by rodrigopaulino — APPROVED** [review]
LGTM!

**Comment by rodrigopaulino** [comment]
ci:forward

**Pull request closed without merging**